core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -9709,4 +9709,12 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
 
     assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedBigQuery));
   }
+
+  @Test void testBigQueryCastGeometry() {
+    String query = "select cast(null as geometry) "
+        + "from \"foodmart\".\"reserve_employee\" ";
+    final String expected = "SELECT CAST(NULL AS GEOGRAPHY)\n"
```

Comment:
Confirm if this this test case is fine.
As we are not giving any value to it and casting null as geometry.
and we tried some values but it din't work.
Please confirm with someone and talk to @waghvijay regarding the same